### PR TITLE
{Release} Upgrade to Azure CLI 2.28.0

### PR DESCRIPTION
**Pull request by azclibot**
Updated Azure CLI version.

### CODE REVIEW

**Comment by yonzhan** [comment]
Release

File: src/azure-cli/HISTORY.rst [hunk 1]
```diff
@@ -3,6 +3,141 @@
...
+* [BREAKING CHANGE] `az aks nodepool update`: Change rejecting the ability to use max-surge with node-image-only (#19295)
+
+**AKS**
+
+* `az aks install-cli`: Add support for kubelogin darwin/arm64 releases (#19072)
+* Fix incorrectly passed parameter for option `--assign-kubelet-identity` in aks create sub-command (#19157)
+* Upgrade api-version to `2021-07-01` for ACS module (#19245)
+* `az aks create/update`: Add support for private cluster public fqdn feature (#19302)
+* Revert PR #18825: az aks create/update: Add parameter --auto-upgrade-channel to support auto upgrade (with fix) (#19297)
+* `aks create\aks nodepool add`: Add parameter ` --os-sku` to support choosing the underlying container host OS (#19374)
```
Comment (jiasli) [inline src/azure-cli/HISTORY.rst:27]:
```suggestion
* `aks create/aks nodepool add`: Add parameter `--os-sku` to support choosing the underlying container host OS (#19374)
```

File: src/azure-cli/HISTORY.rst [hunk 2]
```diff
@@ -3,6 +3,141 @@
 Release History
 ===============
 
+2.28.0
+++++++
+
+**ACR**
+
+* : `az acr create/update`: Add support for disabling export through `--allow-exports` (#19065)
+* : `az acr`: Bump core api-version to `2021-06-01-preview` from `2020-11-01-preview`. agent_pool, tasks and runs operations unchanged from `2019-06-01-preview` (#19065)
```
Comment (Juliehzl) [inline src/azure-cli/HISTORY.rst:12]:
```suggestion
* `az acr create/update`: Add support for disabling export through `--allow-exports` (#19065)
* `az acr`: Bump core api-version to `2021-06-01-preview` from `2020-11-01-preview`. agent_pool, tasks and runs operations unchanged from `2019-06-01-preview` (#19065)
```

File: src/azure-cli/HISTORY.rst [hunk 3]
```diff
@@ -3,6 +3,141 @@
...
+* `az keyvault private-endpoint-connection list`: Support list mhsm's private endpoint connections (#19163)
+* `az keyvault set-policy`: `--key-permissions` add new option `release` (#19411)
+
+**Network**
+
+* Fix NSG rule creation example mistake (#19242)
+* Add a new command group `az network custom-ip prefix`. (#19218)
+* `az network public-ip`: Add parameter `--ip-address`. (#19218)
+* `network public-ip prefix create`: Add parameter `--custom-ip-prefix-name`. (#19218)
+* Fix'az network dns record-set [record-type] add-record' is not idempotent (#19237)
```
Comment (kairu-ms) [inline src/azure-cli/HISTORY.rst:93]:
```suggestion
* Fix `az network dns record-set [record-type] add-record` is not idempotent (#19237)
```

File: src/azure-cli/HISTORY.rst (hunk 1 above)
Comment (Juliehzl) [inline src/azure-cli/HISTORY.rst:27]:
```suggestion
* `aks create/aks nodepool add`: Add parameter ` --os-sku` to support choosing the underlying container host OS (#19374)
```

File: src/azure-cli/HISTORY.rst [hunk 4]
```diff
@@ -3,6 +3,141 @@
...
+* `az role assignment create`: Do not invoke Graph API if `--assignee-principal-type` is provided (#19219)
+
+**SQL**
+
+* `az sql mi update`: Add --subnet and --vnet-name parameters to support the cross subnet update SLO (#18886)
+* Fix the enum name change in track2 Python SDK (#19142)
+
+**Storage**
+
+* Fix #10765 to refine error message when account key is incorrect padding (#13965)
```
Comment (evelyn-ys) [inline src/azure-cli/HISTORY.rst:131]:
```suggestion
* Fix #10765: Refine error message when account key is incorrect padding (#13965)
```

File: src/azure-cli/HISTORY.rst [hunk 5]
```diff
@@ -3,6 +3,141 @@
...
+* :  az webapp config set : Add additional help examples for powershell users (#19185)
+* : Fix #17818: `az functionapp update`:  Add instance validation for updating functionapp (#19141)
+* `az webapp config hostname add`: Fix the issue caused by AttributeError (#19188)
+* `az webapp config hostname add`: Fix the issue caused by AttributeError (#19348)
+* : Fix #16470: `az staticwebapp secrets`: Add commands to manage deployment secrets (#19347)
+* : `az webapp deployment source config-local-git`: Fix the issue caused by AttributeError when slot option is specified (#19404)
+* : `az webapp deleted restore` : Fix the issue that 'WebAppsOperations' object has no attribute 'restore_from_deleted_app' (#19404)
+* `az webapp up`: Add ability to deploy Linux and Windows webapps to the same resource group (#19344)
+* `az webapp up`: Add support for deploying to an App Service Environment (#19425)
+* : Fix #19098: `az webapp deployment slot auto-swap `: Fix the AttributeError error for parameters `--slot --disable` (#19376)
```
Comment (Juliehzl) [inline src/azure-cli/HISTORY.rst:50]:
```suggestion
* `az webapp config storage-account list/add/update/delete`: Remove preview flag (#19177)
* Fix #18497: `functionapp identity show`: Fix the crashes when the functionapp name does not reference an existing functionapp (#19127)
* `az webapp config set`: Add additional help examples for powershell users (#19185)
* Fix #17818: `az functionapp update`:  Add instance validation for updating functionapp (#19141)
* `az webapp config hostname add`: Fix the issue caused by AttributeError (#19188)
* `az webapp config hostname add`: Fix the issue caused by AttributeError (#19348)
* Fix #16470: `az staticwebapp secrets`: Add commands to manage deployment secrets (#19347)
* `az webapp deployment source config-local-git`: Fix the issue caused by AttributeError when slot option is specified (#19404)
* `az webapp deleted restore` : Fix the issue that 'WebAppsOperations' object has no attribute 'restore_from_deleted_app' (#19404)
* `az webapp up`: Add ability to deploy Linux and Windows webapps to the same resource group (#19344)
* `az webapp up`: Add support for deploying to an App Service Environment (#19425)
* : Fix #19098: `az webapp deployment slot auto-swap `: Fix the AttributeError error for parameters `--slot --disable` (#19376)
```

File: src/azure-cli/HISTORY.rst [hunk 6]
```diff
@@ -3,6 +3,141 @@
...
+
+* `az feature registration`: Add az feature registration apis (#19146)
+* `az tag create`: Add the note for handling existing tag in help (#19057)
+* `az ts create`: Fix issue where creating a template spec with inner deployments that reference a common template fails (#19312)
+
+**CDN**
+
+* `az cdn endpoint create`: Fix endpoint creation failure with `--content-types-to-compress` (#19401)
+
+**Compute **
```
Comment (zhoxing-ms) [inline src/azure-cli/HISTORY.rst:62]:
```suggestion
**Compute**
```

File: src/azure-cli/HISTORY.rst [hunk 7]
```diff
@@ -3,6 +3,141 @@
...
+* `az sql mi update`: Add --subnet and --vnet-name parameters to support the cross subnet update SLO (#18886)
+* Fix the enum name change in track2 Python SDK (#19142)
+
+**Storage**
+
+* Fix #10765 to refine error message when account key is incorrect padding (#13965)
+
+**Synapse**
+
+* [BREAKING CHANGE] az synapse workspace key activate: Make command more clearer (#19304)
```
Comment (kairu-ms) [inline src/azure-cli/HISTORY.rst:135]:
```suggestion
* [BREAKING CHANGE] `az synapse workspace key activate`: Make command more clearer (#19304)
```

File: src/azure-cli/HISTORY.rst [hunk 8]
```diff
@@ -3,6 +3,141 @@
...
+
+* `az cdn endpoint create`: Fix endpoint creation failure with `--content-types-to-compress` (#19401)
+
+**Compute **
+
+* `az ssh vm`: Raise error for managed identity and Cloud Shell (#19051)
+* Upgrade api-version for VM and VMSS from `2021-03-01` to `2021-04-01` (#19158)
+* `az vmss create/update`: Support spot restore policy to VM scale sets (#19189)
+* Add new examples for creating disk from share image gallery (#19270)
+* `az vm image​ list\list-offers\list-skus\list-publishers\show`: Add new parameter ​`--edge-zone`​ to support querying the image under edge zone (#19206)
```
Comment (zhoxing-ms) [inline src/azure-cli/HISTORY.rst:68]:
```suggestion
* `az vm image​ list/list-offers/list-skus/list-publishers/show`: Add new parameter ​`--edge-zone`​ to support querying the image under edge zone (#19206)
```

File: src/azure-cli/HISTORY.rst [hunk 9]
```diff
@@ -3,6 +3,141 @@
...
+
+* `az keyvault private-endpoint-connection list`: Support list mhsm's private endpoint connections (#19163)
+* `az keyvault set-policy`: `--key-permissions` add new option `release` (#19411)
+
+**Network**
+
+* Fix NSG rule creation example mistake (#19242)
+* Add a new command group `az network custom-ip prefix`. (#19218)
+* `az network public-ip`: Add parameter `--ip-address`. (#19218)
+* `network public-ip prefix create`: Add parameter `--custom-ip-prefix-name`. (#19218)
```
Comment (Juliehzl) [inline src/azure-cli/HISTORY.rst:92]:
```suggestion
* `az network public-ip prefix create`: Add parameter `--custom-ip-prefix-name`. (#19218)
```

File: src/azure-cli/HISTORY.rst [hunk 10]
```diff
@@ -3,6 +3,141 @@
...
+* Fix the enum name change in track2 Python SDK (#19142)
+
+**Storage**
+
+* Fix #10765 to refine error message when account key is incorrect padding (#13965)
+
+**Synapse**
+
+* [BREAKING CHANGE] az synapse workspace key activate: Make command more clearer (#19304)
+* Code owner change (#19119)
```
Comment (evelyn-ys) [inline src/azure-cli/HISTORY.rst]:
```suggestion
* [BREAKING CHANGE] `az synapse workspace key activate`: Make command clearer (#19304)
```

File: src/azure-cli/HISTORY.rst [hunk 11]
```diff
@@ -3,6 +3,141 @@
...
+* Fix the enum name change in track2 Python SDK (#19142)
+
+**Storage**
+
+* Fix #10765 to refine error message when account key is incorrect padding (#13965)
+
+**Synapse**
+
+* [BREAKING CHANGE] `az synapse workspace key activate`: Make command more clearer (#19304)
+* Code owner change (#19119)
```
Comment (evelyn-ys) [inline src/azure-cli/HISTORY.rst:136]:
```suggestion
* [BREAKING CHANGE] Rename `az synapse workspace key update` to `az synapse workspace key activate` and remove `--is-active` (#19304)
```

**Review by evelyn-ys — APPROVED** [review]

**Review by jiasli — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

File: src/azure-cli/HISTORY.rst [hunk 12]
```diff
@@ -3,6 +3,140 @@
...
+* `az webapp config set`: Add additional help examples for powershell users (#19185)
+* Fix #17818: `az functionapp update`:  Add instance validation for updating functionapp (#19141)
+* `az webapp config hostname add`: Fix the issue caused by AttributeError (#19188)
+* `az webapp config hostname add`: Fix the issue caused by AttributeError (#19348)
+* Fix #16470: `az staticwebapp secrets`: Add commands to manage deployment secrets (#19347)
+* `az webapp deployment source config-local-git`: Fix the issue caused by AttributeError when slot option is specified (#19404)
+* `az webapp deleted restore` : Fix the issue that 'WebAppsOperations' object has no attribute 'restore_from_deleted_app' (#19404)
+* `az webapp up`: Add ability to deploy Linux and Windows webapps to the same resource group (#19344)
+* `az webapp up`: Add support for deploying to an App Service Environment (#19425)
+* : Fix #19098: `az webapp deployment slot auto-swap `: Fix the AttributeError error for parameters `--slot --disable` (#19376)
```
Comment (Juliehzl) [inline src/azure-cli/HISTORY.rst:50]:
```suggestion
* Fix #19098: `az webapp deployment slot auto-swap `: Fix the AttributeError error for parameters `--slot --disable` (#19376)
```

**Review by calvinhzy — APPROVED** [review]

**Review by Juliehzl — APPROVED** [review]

File: src/azure-cli/HISTORY.rst [hunk 13]
```diff
@@ -3,6 +3,140 @@
...
+* Fix incorrectly passed parameter for option `--assign-kubelet-identity` in aks create sub-command (#19157)
+* Upgrade api-version to `2021-07-01` for ACS module (#19245)
+* `az aks create/update`: Add support for private cluster public fqdn feature (#19302)
+* Revert PR #18825: az aks create/update: Add parameter --auto-upgrade-channel to support auto upgrade (with fix) (#19297)
+* `aks create/aks nodepool add`: Add parameter ` --os-sku` to support choosing the underlying container host OS (#19374)
+
+**AMS**
+
+* Update python to track 2 (#18687)
+
```
Comment (jsntcy) [inline src/azure-cli/HISTORY.rst:32]:
```suggestion
```

**Review by kairu-ms — APPROVED** [review]

File: src/azure-cli/HISTORY.rst [hunk 14]
```diff
@@ -3,6 +3,140 @@
...
+* [BREAKING CHANGE] `az aks nodepool update`: Change rejecting the ability to use max-surge with node-image-only (#19295)
+
+**AKS**
+
+* `az aks install-cli`: Add support for kubelogin darwin/arm64 releases (#19072)
+* Fix incorrectly passed parameter for option `--assign-kubelet-identity` in aks create sub-command (#19157)
+* Upgrade api-version to `2021-07-01` for ACS module (#19245)
+* `az aks create/update`: Add support for private cluster public fqdn feature (#19302)
+* Revert PR #18825: az aks create/update: Add parameter --auto-upgrade-channel to support auto upgrade (with fix) (#19297)
```
Comment (zhoxing-ms) [inline src/azure-cli/HISTORY.rst:26]:
```suggestion
* Revert PR #18825: `az aks create/update`: Add parameter `--auto-upgrade-channel` to support auto upgrade (with fix) (#19297)
```

**Pull request merged**